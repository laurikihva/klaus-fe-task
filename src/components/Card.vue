<template>
  <div class="bg-white rounded-lg px-4 py-6">
    <slot />
  </div>
</template>
