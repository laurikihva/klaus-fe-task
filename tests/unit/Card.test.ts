import { mount } from "@vue/test-utils";

import Card from "@/components/Card.vue";

const wrapper = mount(Card, {
  slots: {
    default: "Card content",
  },
});

describe("Loads and displays Card component", () => {
  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("contains slot content and base classes", () => {
    expect(wrapper.text()).toContain("Card content");

    expect(wrapper.classes("bg-white")).toBe(true);
    expect(wrapper.classes("rounded-lg")).toBe(true);
    expect(wrapper.classes("px-4")).toBe(true);
    expect(wrapper.classes("py-6")).toBe(true);
  });
});
