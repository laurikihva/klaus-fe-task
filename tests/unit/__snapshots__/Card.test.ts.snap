// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loads and displays Card component > renders correctly 1`] = `"<div class="bg-white rounded-lg px-4 py-6">Card content</div>"`;
